### PR TITLE
Migrate String Literal Input Component

### PR DESCRIPTION
**Pull request by mdelez**
closes https://dasch.myjetbrains.com/youtrack/issue/DSP-491

### CODE REVIEW

**Comment by mdelez** [comment]
@kilchenmann I noticed that with the simple input version, if you click on the language dropdown button and then don't click on a language but rather click somewhere else to close the menu, the input will remain disabled. Do you have an idea on how to refactor this so that the input becomes enabled anytime the language dropdown is closed?

**Comment by kilchenmann** [comment]
> @kilchenmann I noticed that with the simple input version, if you click on the language dropdown button and then don't click on a language but rather click somewhere else to close the menu, the input will remain disabled. Do you have an idea on how to refactor this so that the input becomes enabled anytime the language dropdown is closed?

Oh, that's a weird issue. I have to look at the code to understand what happens when...

**Comment by mdelez** [comment]
> > @kilchenmann I noticed that with the simple input version, if you click on the language dropdown button and then don't click on a language but rather click somewhere else to close the menu, the input will remain disabled. Do you have an idea on how to refactor this so that the input becomes enabled anytime the language dropdown is closed?
> 
> Oh, that's a weird issue. I have to look at the code to understand what happens when...

It comes from the click event on the button that opens the language select button
https://github.com/dasch-swiss/dsp-ui-lib/blob/2c79bf0aeb5d51a03e74223c983cb4d7bd3af174/projects/dsp-ui/src/lib/action/components/string-literal-input/string-literal-input.component.html#L7-L17

However, the method to re-enable the input (switchFocus()) is only called when one of the mat-menu-items is selected
https://github.com/dasch-swiss/dsp-ui-lib/blob/2c79bf0aeb5d51a03e74223c983cb4d7bd3af174/projects/dsp-ui/src/lib/action/components/string-literal-input/string-literal-input.component.html#L20-L28

https://github.com/dasch-swiss/dsp-ui-lib/blob/2c79bf0aeb5d51a03e74223c983cb4d7bd3af174/projects/dsp-ui/src/lib/action/components/string-literal-input/string-literal-input.component.ts#L180-L190

**Comment by mdelez** [comment]
> > > @kilchenmann I noticed that with the simple input version, if you click on the language dropdown button and then don't click on a language but rather click somewhere else to close the menu, the input will remain disabled. Do you have an idea on how to refactor this so that the input becomes enabled anytime the language dropdown is closed?
> > 
> > 
> > Oh, that's a weird issue. I have to look at the code to understand what happens when...
> 
> It comes from the click event on the button that opens the language select button
> https://github.com/dasch-swiss/dsp-ui-lib/blob/2c79bf0aeb5d51a03e74223c983cb4d7bd3af174/projects/dsp-ui/src/lib/action/components/string-literal-input/string-literal-input.component.html#L7-L17
> 
> However, the method to re-enable the input (switchFocus()) is only called when one of the mat-menu-items is selected

@kilchenmann fixed in e276957 :)

File: src/app/action-playground/action-playground.component.html [hunk 1]
```diff
@@ -66,3 +66,13 @@
 
 <strong>Show only one value (Uses the language of the user if logged in. If not, uses the browser language)</strong>
 <p>{{labels | dspStringifyStringLiteral}}</p>
+
+<hr>
+<p>StringLiteral input example</p>
+<dsp-string-literal-input [placeholder]="'List label'"> </dsp-string-literal-input>
```
Comment (kilchenmann) [inline src/app/action-playground/action-playground.component.html:72]: Can you add the example with the output here and display the values as console.log? 
e.g.

```typescript
label: StringLiteral[];
...
handleData(data: StringLiteral[]) {
  this.label = data;
}

submitData() {
  console.log('submit string literal', this.label);
}
```

```html
<dsp-string-literal-input [placeholder]="'List label'" (dataChanged)="handleData($event)" (enter)="submitData()""></dsp-string-literal-input>

File: projects/dsp-ui/src/lib/action/components/string-literal-input/string-literal-input.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import { MatButtonToggleModule } from '@angular/material/button-toggle';
+import { MatButtonHarness } from '@angular/material/button/testing';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatInputModule } from '@angular/material/input';
+import { MatInputHarness } from '@angular/material/input/testing';
+import { MatMenuModule } from '@angular/material/menu';
+import { MatMenuHarness, MatMenuItemHarness } from '@angular/material/menu/testing';
```
Comment (flavens) [inline projects/dsp-ui/src/lib/action/components/string-literal-input/string-literal-input.component.spec.ts:13]: the unused imports should be removed

File: src/app/action-playground/action-playground.component.html (hunk 1 above)
Comment (mdelez) [reply src/app/action-playground/action-playground.component.html]: added in ae0538c

File: projects/dsp-ui/src/lib/action/components/string-literal-input/string-literal-input.component.spec.ts (hunk 2 above)
Comment (kilchenmann) [reply projects/dsp-ui/src/lib/action/components/string-literal-input/string-literal-input.component.spec.ts:13]: That's a good point. @mdelez please use the `shift+alt+o`-command to organize imports in vsc.

Comment (mdelez) [reply projects/dsp-ui/src/lib/action/components/string-literal-input/string-literal-input.component.spec.ts]: done in 7276257, pretty nifty command :)

**Pull request merged**